### ISSUE TITLE
results do not meet the indicators in the paper

### ISSUE THREAD

**Issue by wqqqqqqw** [TLDBN/SSLFusion#1]
Hello, thank you for your excellent work. I reproduced the code but the results do not meet the indicators in the paper. I set the batch size from 4 to 2 and adjusted the learning rate from 0.02 to 0.01 accordingly, training on 4 rtx3090. I did not change anything else. Has anyone else been able to reproduce the indicators from the paper? I am not sure what the reason is.
`2025-09-24 10:34:55,155   INFO  Car AP@0.70, 0.70, 0.70:                                                                                                                                
bbox AP:90.6440, 89.2376, 89.1693                                                                                                                                                       
bev  AP:89.9691, 87.6571, 87.6653                                                                                                                                                       
3d   AP:89.2831, 79.0446, 78.8168                                                                                                                                                       
aos  AP:90.63, 89.16, 89.03                                                                                                                                                             
Car AP_R40@0.70, 0.70, 0.70:                                                                                                                                                            
bbox AP:96.4337, 94.0595, 92.3295                                                                                                                                                       
bev  AP:93.2502, 88.7695, 88.8792                                                                                                                                                       
3d   AP:92.2994, 82.9796, 82.8340                                                                                                                                                       
aos  AP:96.41, 93.94, 92.16                                                                                                                                                             
Car AP@0.70, 0.50, 0.50:                                                                                                                                                                
bbox AP:90.6440, 89.2376, 89.1693                                                                                                                                                       
bev  AP:90.6125, 89.2053, 89.1614                                                                                                                                                       
3d   AP:90.6125, 89.1921, 89.1428                                                                                                                                                       
aos  AP:90.63, 89.16, 89.03                                                                                                                                                             
Car AP_R40@0.70, 0.50, 0.50:                                                                                                                                                            
bbox AP:96.4337, 94.0595, 92.3295                                                                                                                                                       
bev  AP:96.4157, 94.2845, 94.4209                                                                                                                                                       
3d   AP:96.3971, 94.2523, 94.3172                                                                                                                                                       
aos  AP:96.41, 93.94, 92.16                                                                                                                                                             
Pedestrian AP@0.50, 0.50, 0.50:                                                                                                                                                         
bbox AP:76.6922, 73.4358, 69.4069                                                                                                                                                       
bev  AP:67.3485, 63.4021, 58.3728                                                                                                                                                       
3d   AP:65.1767, 60.5504, 56.1601                                                                                                                                                       
aos  AP:70.91, 67.19, 63.38                                                                                                                                                             
Pedestrian AP_R40@0.50, 0.50, 0.50:                                                                                                                                                     
bbox AP:77.6299, 73.7749, 71.1969                                                                                                                                                       
bev  AP:68.4973, 62.9362, 58.8041                                                                                                                                                       
3d   AP:64.9333, 59.4066, 55.4110                                                                                                                                                       
aos  AP:71.49, 67.25, 64.50                                                                                                                                                             
Pedestrian AP@0.50, 0.25, 0.25:                                                                                                                                                         
bbox AP:76.6922, 73.4358, 69.4069                                                                                                                                                       
bev  AP:79.1067, 76.3752, 74.3123                                                                                                                                                       
3d   AP:79.0675, 76.3327, 74.1838                                                                                                                                                       
aos  AP:70.91, 67.19, 63.38        
Pedestrian AP_R40@0.50, 0.25, 0.25:
bbox AP:77.6299, 73.7749, 71.1969
bev  AP:81.5343, 77.8495, 74.9064
3d   AP:81.4283, 77.7681, 74.7796
aos  AP:71.49, 67.25, 64.50
Cyclist AP@0.50, 0.50, 0.50:     
bbox AP:88.5573, 80.0628, 76.2682
bev  AP:87.1543, 76.8017, 72.8494
3d   AP:86.0975, 72.6696, 70.5974
aos  AP:88.31, 77.59, 74.10
Cyclist AP_R40@0.50, 0.50, 0.50:
bbox AP:93.5243, 80.6955, 78.6511
bev  AP:91.8043, 77.0192, 74.6120
3d   AP:90.5655, 74.2714, 71.8661
aos  AP:93.18, 78.16, 76.23
Cyclist AP@0.50, 0.25, 0.25:
bbox AP:88.5573, 80.0628, 76.2682
bev  AP:87.5516, 77.3004, 76.4680
3d   AP:87.5516, 77.3004, 76.4680
aos  AP:88.31, 77.59, 74.10
Cyclist AP_R40@0.50, 0.25, 0.25:
bbox AP:93.5243, 80.6955, 78.6511
bev  AP:92.3279, 77.7952, 76.5815
3d   AP:92.3279, 77.7952, 76.5815
aos  AP:93.18, 78.16, 76.23

2025-09-24 10:34:55,158   INFO  Result is saved to /data/wanqiang_space/detection/SSLFusion/output/cfgs/kitti/voxel_rcnn_3class_SSLFusion/default/eval/eval_with_train/epoch_80/val
2025-09-24 10:34:55,158   INFO  ****************Evaluation done.*****************
2025-09-24 10:34:55,183   INFO  Epoch 80 has been evaluated
2025-09-24 10:35:25,214   INFO  **********************End evaluation cfgs/kitti/voxel_rcnn_3class_SSLFusion(default)**********************SLFusion/default/ckpt `

**Comment by TLDBN** [comment]
Hi, sorry for the late reply. Could you please try to train the model on a single class, like Car, to see the results?

**Comment by wqqqqqqw** [comment]
Hi, I have trained the model on single Car class, but the results are still not good.
2025-11-23 17:15:47,174   INFO  **********************Start evaluation cfgs/kitti/voxel_rcnn_1class_SSLFusion(default)**********************
2025-11-23 17:15:47,175   INFO  Loading KITTI dataset
2025-11-23 17:15:47,308   INFO  Total samples for KITTI dataset: 3769
2025-11-23 17:15:47,310   INFO  ==> Loading parameters from checkpoint /data/wanqiang_space/detection/SSLFusion/output/cfgs/kitti/voxel_rcnn_1class_SSLFusion/default/ckpt/checkpoint_epoch_80.pth to CPU
2025-11-23 17:15:47,517   INFO  ==> Checkpoint trained from version: pcdet+0.6.0+8fe3cb7
2025-11-23 17:15:47,627   INFO  ==> Done (loaded 1132/1132)
2025-11-23 17:15:47,636   INFO  *************** EPOCH 80 EVALUATION *****************
2025-11-23 17:17:57,786   INFO  *************** Performance of EPOCH 80 *****************
2025-11-23 17:17:57,786   INFO  Generate label finished(sec_per_example: 0.0345 second).
2025-11-23 17:17:57,786   INFO  recall_roi_0.3: 0.969556
2025-11-23 17:17:57,786   INFO  recall_rcnn_0.3: 0.969556
2025-11-23 17:17:57,787   INFO  recall_roi_0.5: 0.963092
2025-11-23 17:17:57,787   INFO  recall_rcnn_0.5: 0.963717
2025-11-23 17:17:57,787   INFO  recall_roi_0.7: 0.826023
2025-11-23 17:17:57,787   INFO  recall_rcnn_0.7: 0.869257
2025-11-23 17:17:57,788   INFO  Average predicted number of objects(3769 samples): 5.318
2025-11-23 17:18:15,331   INFO  Car AP@0.70, 0.70, 0.70:
bbox AP:97.6898, 89.6245, 89.4494
bev  AP:90.3355, 88.5909, 88.3950
3d   AP:89.6945, 84.9752, 79.3014
aos  AP:97.67, 89.55, 89.33
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:98.5314, 94.9921, 94.8801
bev  AP:95.9830, 91.7042, 91.4951
3d   AP:92.8926, 85.6567, 83.5539
aos  AP:98.51, 94.89, 94.70
Car AP@0.70, 0.50, 0.50:
bbox AP:97.6898, 89.6245, 89.4494
bev  AP:97.7370, 89.6051, 89.4548
3d   AP:97.6985, 89.5868, 89.4289
aos  AP:97.67, 89.55, 89.33
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:98.5314, 94.9921, 94.8801
bev  AP:98.5425, 95.0404, 95.0111
3d   AP:98.5204, 94.9974, 94.9546
aos  AP:98.51, 94.89, 94.70

2025-11-23 17:18:15,331   INFO  Result is saved to /data/wanqiang_space/detection/SSLFusion/output/cfgs/kitti/voxel_rcnn_1class_SSLFusion/default/eval/eval_with_train/epoch_80/val
2025-11-23 17:18:15,331   INFO  ****************Evaluation done.*****************
2025-11-23 17:18:15,353   INFO  Epoch 80 has been evaluated
2025-11-23 17:18:45,385   INFO  **********************End evaluation cfgs/kitti/voxel_rcnn_1class_SSLFusion(default)**********************

**Comment by TLDBN** [comment]
Hi, could you please try to evaluate checkpoints of epoch 70~80 and see the performance? Because we used early-stop to get the best results, and you need to train more than once to get the best results, thanks! And if you can, please set everything the same as the paper report, like batch size, as it may influence the performance.

**Comment by wqqqqqqw** [comment]
Okay, thank you for your quick response.
